keras/backend/tensorflow_backend.py

```diff
@@ -367,10 +712,47 @@ def moving_average_update(variable, value, momentum):
 # LINEAR ALGEBRA
 
 def dot(x, y):
-    '''Multiplies 2 tensors.
+    '''Multiplies 2 tensors and/or variables.
     When attempting to multiply a ND tensor
     with a ND tensor, reproduces the Theano behavior
     (e.g. (2, 3).(4, 3, 5) = (2, 4, 5))
+
+    # Arguments
+        x: Keras variable.
+        y: Keras variable.
+
+    # Returns
+        Dot product of `x` and `y`.
+
+    # Examples
+    ```python
+        # dot product between Keras variables
+        >>> x = K.random_uniform_variable(shape=(2, 3), low=0, high=1)
+        >>> y = K.random_uniform_variable(shape=(3, 4), low=0, high=1)
+        >>> xy = K.dot(x, y)
+        >>> K.int_shape(xy)
+        (2, 4)
+        >>> K.eval(xy)
+        array([[ 0.36257121,  0.450986  ,  0.49147484,  0.20764332],
+               [ 0.45370784,  1.53120756,  1.37138271,  0.24922732]], dtype=float32)
+    ```
+
+    ```python
+        # dot product between Keras tensor and Keras variable
```

Comment:
Not a useful example. Instead please have examples dealing with larger input shapes (e.g. 3D, 4D).